### ISSUE TITLE
OIDC/SSO/Oauth2 support

### ISSUE THREAD

**Issue by SadmL** [fedify-dev/hollo#69]
Does Hollo has OIDC/SSO/Oauth2 support?
Can't find any info about it.
I understand that project/software single-user first, but...

~~If not implemented and not interested in, feel free to close this issue.~~

**Comment by dahlia** [comment]
It's not implemented yet, but it is on our roadmap.

**Comment by ThisIsMissEm** [comment]
What would the goal of OIDC for SSO be? Hollo would still need to be an OAuth 2 authorization server to implement the Mastodon API (which isn't a specification but an implementation)

Typically SSO only makes a tonne of sense in environments where you support multiple users, otherwise it's just a whole heap of security attack surface.

**Comment by SadmL** [comment]
Just wanted single (almost) click login 🙃
I'm using Nextcloud as OIDC issuer via https://github.com/H2CK/oidc.
Currently using GoToSocial [that has OIDC support](https://docs.gotosocial.org/en/latest/configuration/oidc), just wanted try Hollo, but I'll wait when this would implemented (no rush)

**Comment by ThisIsMissEm** [comment]
Yeah, GoToSocial is a multi account server, so OIDC makes sense there. In Hollo, there is only one account with a login, which has access to a number of virtual accounts.

So I'm not sure I see how OIDC would work here? You'd have to like restrict access to a specific `sub` (IdP account) or to specific roles or something.

Adding this would increase the attack surface available quite significantly, and it doesn't really seem inline with Hollo's goal of being a "Federated **single-user** microblogging server"
